File: user-service/src/main/java/io/github/tdminhnhat/entity/Address.java
package io.github.tdminhnhat.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.*;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

@Entity @Table(name = "addresses")
@Getter @Setter
@NoArgsConstructor
@DynamicInsert @DynamicUpdate
public class Address extends BaseEntityJPA {

    @Column(name = "address", length = 50)
    private String address;

    @Column(name = "ward", length = 50)
    private String ward;

    @Column(name = "city", length = 50)
    private String city;

    @Column(name = "country", length = 50)
    private String country;

    @Column(name = "zip_code", length = 50)
    private String zipCode;

    @OneToOne(mappedBy = "address")
    private User user;

    public Address(String address, String ward, String city, String country, String zipCode) {
        this.address = address;
        this.ward = ward;
        this.city = city;
        this.country = country;
        this.zipCode = zipCode;
    }
}

File: jeri-chat-mobile/lib/model/entity/address.dart

class Address {
  String _address;
  String _ward;
  String _city;
  String _country;
  String _zipCode;

  Address(this._address, this._ward, this._city, this._country, this._zipCode);

  String get zipCode => _zipCode;

  set zipCode(String value) {
    _zipCode = value;
  }

  String get country => _country;

  set country(String value) {
    _country = value;
  }

  String get city => _city;

  set city(String value) {
    _city = value;
  }

  String get ward => _ward;

  set ward(String value) {
    _ward = value;
  }

  String get address => _address;

  set address(String value) {
    _address = value;
  }
}